src/pages/index.js

```diff
@@ -136,28 +125,110 @@ const editFormValidator = new FormValidator(
 );
 const addCardFormValidator = new FormValidator(validationSettings, addCardForm);
 
-editFormValidator.enableValidation();
+const editAvatarValidator = new FormValidator(
+  validationSettings,
+  avatarEditForm
+);
 
+editFormValidator.enableValidation();
 addCardFormValidator.enableValidation();
+editAvatarValidator.enableValidation();
+
+/* -------------------------------------------------------------------------- */
+/*                      Project 8 GENERATE INITIAL CARDS                      */
+/* -------------------------------------------------------------------------- */
+
+// Preview image Card
+const previewimagePopup = new PopupWithImage("#preview-image-modal");
 
 /* -------------------------------------------------------------------------- */
-/*                                  Project 8                                 */
+/*                                  Project 9                                 */
 /* -------------------------------------------------------------------------- */
 
-// Edit Profile Info
-const userInfo = new UserInfo({
+const api = new Api({
+  url: "https://around.nomoreparties.co/v1/cohort-3-en",
+  headers: {
+    authorization: "b32399ae-a567-415e-9d15-bc2048a1a730",
+    "Content-Type": "application/json",
+  },
+  response: (res) => {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}`);
+  },
+});
+
+// 1. Loading user information from the server
+api
+  .getUserInfo()
+  .then((user) => {
+    userInfo.setUserInfo({ name: user.name, about: user.about });
+    userID = user._id;
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+// 2. Loading cards from the server
+
+api
+  .getInitialCards()
+  .then((initialCards) => {
+    cardList = new Section(
+      {
+        items: initialCards,
+        renderer: renderCard,
+      },
+      ".card__list"
+    );
+
+    cardList.renderItems();
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+// 3. Editing the profile
+let userInfo = new UserInfo({
   userNameSelector: ".profile__title",
   userDescriptionSelector: ".profile__description",
 });
 
 const profileEditPopup = new PopupWithForm(
   "#profile-edit-modal",
   (inputValues) => {
+    profileEditFormButton.textContent = "Saving...";
```

Comment:
The code for changing the button text should be implemented as a separate function or method in `PopupWithForm` and `PopupWithConfirmation` classes

One of the options to implement that would be adding a parameter `loadingButtonText` to the constructor of `PopupWithForm` and using it to change button text. 

```js
constructor(popupSelector, { handleSubmit, loadingButtonText }) { 
  ...
  this._submitButton = this._formElement.querySelector(...)
  this._buttonText = this._submitButton.textContent;
  this._loadingButtonText = loadingButtonText;
  …
}

showLoading() {
  this._submitButton.textContent = this._loadingButtonText
}

hideLoading() {
  this._submitButton.textContent = this._buttonText
}
``` 

The api call should look like this:

```js
popup.showLoading()
api.
  .then(...)
  .catch(...)
  .finally(() => {
    popup.hideLoading()